server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1427,18 +1429,18 @@ static void verifyChecksum(BufferedChecksumStreamInput in) throws IOException {
         long expectedChecksum = in.getChecksum();
         long readChecksum = Integer.toUnsignedLong(in.readInt());
         if (readChecksum != expectedChecksum) {
-            throw new TranslogCorruptedException("translog stream is corrupted, expected: 0x" +
+            throw new TranslogCorruptedException(in.getSource(), "checksum verification failed - expected: 0x" +
```

Comment:
Could you restore the `translog stream is corrupted` content of this message?

Reply:
Here you will have the following message: translog from source [...] is corrupted, checksum verification failed .... What is the reason to restore the original message? IMHO, this one is more precise

Reply:
Ok. Changing messages like this can cause a hassle when trying to diagnose issues, because it breaks the "grep the code for the error message in approximately the right version of the code" workflow, but in this case I think the new message is a sufficiently large improvement.